### ISSUE TITLE
--seed does not change the fintuning results of the xlnet model

### ISSUE THREAD

**Issue by zhaoguangxiang** [huggingface/pytorch-transformers#1160]
## 🐛 Bug

Model I am using  (XLNet):

Language I am using the model on (English):

The problem arise when using:
* [ ] the official example scripts
gpu=3
seed=1
task=MRPC
bsz=32
learning_rate=5e-5
max_steps=800
warmup_steps=200
save_steps=400
export CUDA_VISIBLE_DEVICES=${gpu}
export GLUE_DIR=/home/zhaoguangxiang/bert/glue_data
python3 ./examples/run_glue.py \
    --model_type xlnet \
    --model_name_or_path xlnet-large-cased \
    --do_train  \
    --do_eval   \
    --task_name=${task}     \
    --data_dir=${GLUE_DIR}/${task}  \
    --output_dir=checkpoint/xl_${task}_seed${seed}/   \
    --max_seq_length=128   \
    --per_gpu_eval_batch_size=${bsz}   \
    --per_gpu_train_batch_size=${bsz}   \
    --gradient_accumulation_steps=1 \
    --max_steps=${max_steps}  \
    --model_name=xlnet-large-cased   \
    --overwrite_output_dir   \
    --overwrite_cache \
    --save_steps ${save_steps} \
    --learning_rate ${learning_rate} \
    --warmup_steps=${warmup_steps}
The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)

## To Reproduce

1. change the seed
2. results does not change, and keep acc=0.8774509803921569 for every seed

## Environment

* OS:linux
* Python version:3.6
* PyTorch version: 1.2
* PyTorch Transformers version (or branch): latest
* Using GPU ?  1 * TITAN RTX

**Comment by thomwolf** [comment]
In the command line you are showing, you should add a `--seed ${seed}` argument to set the seed, otherwise, it will stay the same.

**Comment by zhaoguangxiang** [comment]
> In the command line you are showing, you should add a --seed ${seed} argument to set the seed, otherwise, it will stay the same.

Sorry, i forgot it.
